forge/ee/routes/flowBlueprints/index.js

```diff
@@ -243,6 +243,94 @@ module.exports = async function (app) {
         }
     })
 
+    app.get('/export', {
+        preHandler: app.needsPermission('flow-blueprint:create'),
+        schema: {
+            summary: 'Export one or more Blueprints',
+            tags: ['Flow Blueprints'],
+            query: {
+                id: { type: 'array', items: { type: 'string' } }
+            },
+            response: {
+                200: {
+                    type: 'object',
+                    allOf: [{ $ref: 'FlowBlueprintExport' }],
+                    properties: {
+                        count: { type: 'integer' }
+                    }
+                },
+                '4xx': {
+                    $ref: 'APIError'
+                }
+            }
+        }
+    }, async (request, reply) => {
+        let where = {}
+        if (request.query.id && typeof request.query.id === 'string') {
+            where = { id: app.db.models.FlowTemplate.decodeHashid(request.query.id)[0] }
+        } else if (request.query.id && Array.isArray(request.query.id)) {
+            where = { id: request.query.id.map(i => app.db.models.FlowTemplate.decodeHashid(i)[0]) }
+        }
+        const flowTemplates = await app.db.models.FlowTemplate.getAll({}, where)
+        reply.send({
+            blueprints: flowTemplates.templates.map(bp => app.db.views.FlowTemplate.flowBlueprintExport(bp)),
+            count: flowTemplates.templates.length
+        })
+    })
+
+    app.post('/import', {
+        preHandler: app.needsPermission('flow-blueprint:create'),
+        schema: {
+            summary: 'Import one or more Blueprints',
+            tags: ['Flow Blueprints'],
+            body: {
+                type: 'object',
+                allOf: [{ $ref: 'FlowBlueprintExport' }]
+            },
+            response: {
+                201: {
+                    type: 'object',
+                    properties: {
+                        blueprints: { type: 'array', items: { $ref: 'FlowBlueprintSummary' } },
+                        count: { type: 'integer' }
+                    }
+                },
+                '4xx': {
+                    $ref: 'APIError'
+                }
+            }
+        }
+    }, async (request, reply) => {
+        const newBlueprints = request.body
+        const existingBlueprints = await app.db.models.FlowTemplate.getAll()
+        const response = {
+            count: 0,
+            blueprints: []
+        }
+        for (const nbp of newBlueprints.blueprints) {
+            // check for clash
+            let clash = false
+            for (const existing of existingBlueprints.templates) {
```

Comment:
My only concern is this nested for loop which is an O(n²) notation. If you don't think it will cause problems it can be ignored

Reply:
I can't think of a better way to do it so as to check all the names for duplicates.

We can't remove them from the "existing" list as there may be multiple matches (but that would still end up with name clashes in the newly generated names).

Reply:
We'll change it if needed